### PR TITLE
Add e2e tests

### LINKED ISSUE #21: Add CI to validate the enterprise chart
We should add a CI check to verify that all components in this chart depend on the same version of the open source Gloo Mesh repo.

### PR DESCRIPTION
**Pull request by ilackarms**
e2e tests are imported from oss and gme repos
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise-helm/issues/21

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      image: ubuntu-1604:201903-01
+      # need to upgrade plan before using large executor
+      resource_class: large
+    working_directory: ~/go/src/github.com/solo-io/gloo-mesh-enterprise-helm
+    steps:
+      - checkout
+      - helm/install-helm-client:
+          version: v3.2.4
+      - run:
+          name: install go
```
Comment (harveyxia) [inline .circleci/config.yml:16]: (not needed for this PR) we could eventually leverage [Circle's caching feature](https://circleci.com/docs/2.0/caching/) to save some compute in these setup steps

File: ci/update_gomod.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"os/exec"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/go-utils/log"
+	"github.com/solo-io/skv2/codegen/util"
+)
+
+// this file is a Go script which udpates the go.mod file to match the contents of Chart-template.yaml
```
Comment (harveyxia) [inline ci/update_gomod.go]:
```suggestion
// this file is a Go script which updates the go.mod file to match the contents of Chart-template.yaml
```

File: ci/update_gomod.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+// the versions of subcharts we import
+type subchartVersions struct {
+	glooMesh           string
+	glooMeshEnterprise string
+}
+
+func (v subchartVersions) updateGoMod() error {
+	for repo, version := range map[string]string{
+		glooMeshRepo:           v.glooMesh,
+		glooMeshEnterpriseRepo: v.glooMeshEnterprise,
```
Comment (harveyxia) [inline ci/update_gomod.go:74]: Given that [enterprise's go mod declares OSS as a dependency](https://github.com/solo-io/gloo-mesh-enterprise/blob/master/go.mod#L36), would it be sufficient to just install the go mod deps for Gloo Mesh enterprise?

File: test/e2e/istio/istio_e2e_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+	return nil
+}
+
+func getApiserverAddress(cluster string) (string, error) {
+	switch runtime.GOOS {
+	case "darwin":
+		return "host.docker.internal", nil
+	default:
+		addr, err := runCommandOut("bash", "-c", `docker exec `+cluster+`-control-plane ip addr show dev eth0 | sed -nE 's|\s*inet\s+([0-9.]+).*|\1|p'`)
```
Comment (harveyxia) [inline test/e2e/istio/istio_e2e_suite_test.go:186]: can we reuse `DockerHostAddress` here?

File: ci/update_gomod.go (hunk 3 above)
Comment (ilackarms) [reply ci/update_gomod.go:74]: we may want changes from OSS that do not require a version bump in enterprise. these e2e tests should verify compatibility

File: test/e2e/istio/istio_e2e_suite_test.go (hunk 4 above)
Comment (ilackarms) [reply test/e2e/istio/istio_e2e_suite_test.go:186]: this isn't getting the address of the docker *host* but the docker *containers* for each kind cluster

**Review by harveyxia — APPROVED** [review]

**Pull request merged**